### PR TITLE
Rancher v2.4 features

### PR DESCRIPTION
**Pull request by rawmind0**
This PR contains:
* Updated go modules and vendor files to rancher v2.4.0
* Updated rancher to v2.4.0 and k3s to v1.17.4-k3s1 on acceptance tests
* Added `group_principal_id` argument to `rancher2_global_role_binding` resource
* Added `k3s_config` argument to `rancher2_cluster` datasource and resource
* Added `version` argument to `rancher2_catalog` datasource and resource
* Added `upgrade_strategy` argument to `rke_config` on `rancher2_cluster` resource
* Added `scheduled_cluster_scan` argument on `rancher2_cluster` and `rancher2_cluster_template` resources
* Added `rancher2_cluster_scan` datasource

### CODE REVIEW

File: website/docs/d/globalRole.html.markdown [hunk 1]
```diff
@@ -27,6 +27,7 @@ data "rancher2_global_role_binding" "foo" {
 ## Attributes Reference
 
 * `id` - (Computed) The ID of the resource (string)
+* `group_principal_id` - (Computed) The group principal ID to assign global role binding. Rancher v2.4.0 or higher is required (string)
```
Comment (cloudnautique) [inline website/docs/d/globalRole.html.markdown:30]: Why can I use `user_id` but here we need `_principal_`

File: website/docs/r/projectRole.html.markdown [hunk 2]
```diff
@@ -36,6 +36,8 @@ The following arguments are supported:
 * `annotations` - (Optional/Computed) Annotations of the resource (map)
 * `labels` - (Optional/Computed) Labels of the resource (map)
 
+**Note** user `user_id | user_principal_id` OR group `group_id | group_principal_id` must be defined
```
Comment (cloudnautique) [inline website/docs/r/projectRole.html.markdown:39]: When would you use user_id or group_id over the principal version?

File: website/docs/r/catalog.html.markdown [hunk 3]
```diff
@@ -47,6 +47,7 @@ The following arguments are supported:
 * `refresh` - (Optional) Catalog will wait for refresh after tf creation and on every tf read. Default `false` (bool)
 * `scope` - (Optional) The scope of the catalog. `cluster`, `global`, and `project` are supported. Default `global` (string)
 * `username` - (Optional/Sensitive) The username to access the catalog if needed (string)
+* `version` - (Optional/ForceNew) Helm version for the catalog. Available options: `helm_v2` (default) and `helm_v3` (string)
```
Comment (cloudnautique) [inline website/docs/r/catalog.html.markdown:50]: Are these the options from the Rancher API?

File: website/docs/r/cluster.html.markdown [hunk 4]
```diff
@@ -698,6 +698,25 @@ The following attributes are exported:
 * `extra_env` - (Optional) Extra environment for scheduler service (list)
 * `image` - (Optional/Computed) Docker image for scheduler service (string)
 
+#### `upgrade_strategy`
+
+##### Arguments
+
+* `drain` - (Optional) RKE drain nodes. Default: `false` (bool)
+* `drain_input` - (Optional/Computed) RKE drain node input (list Maxitems: 1)
```
Comment (cloudnautique) [inline website/docs/r/cluster.html.markdown:786]: What is this?

File: website/docs/r/catalog.html.markdown (hunk 3 above)
Comment (rawmind0) [reply website/docs/r/catalog.html.markdown:50]: Yes, these options comes from Rancher API

File: website/docs/r/cluster.html.markdown (hunk 4 above)
Comment (rawmind0) [reply website/docs/r/cluster.html.markdown:786]: You mean drain_input??

File: website/docs/d/clusterScan.html.markdown [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+Use this data source to retrieve information about a Rancher v2 Cluster CIS Scan resource.
+
+## Example Usage
+
+```hcl
+data "rancher2_cluster_scan" "foo" {
+    cluster_id = <clusterID>
+    name = "foo"
+}
```
Comment (cloudnautique) [inline website/docs/d/clusterScan.html.markdown:19]: Can we show an example of scheduled? I can't see wanting to run an adhoc scan with TF as the norm.

File: website/docs/d/globalRole.html.markdown (hunk 1 above)
Comment (rawmind0) [reply website/docs/d/globalRole.html.markdown:30]: This is how is defined at API, https://github.com/rancher/types/blob/master/client/management/v3/zz_generated_global_role_binding.go#L28

File: website/docs/r/projectRole.html.markdown (hunk 2 above)
Comment (rawmind0) [reply website/docs/r/projectRole.html.markdown:39]: Probably never, but there is the option at API.. :)

File: website/docs/d/clusterScan.html.markdown (hunk 5 above)
Comment (rawmind0) [reply website/docs/d/clusterScan.html.markdown:19]: This is a datasource intended just to get status about already run cluster scans.  cluster_scan resource didn't exist just by the reason you mention.

Comment (rawmind0) [reply website/docs/d/clusterScan.html.markdown:19]: Added examples on cluster and cluster template docs

File: website/docs/d/globalRole.html.markdown (hunk 1 above)
Comment (rawmind0) [reply website/docs/d/globalRole.html.markdown:30]: Added examples on globlaRole resource doc.

**Comment by rawmind0** [comment]
Updated acceptance test to rancher v2.4.0

**Review by cloudnautique — APPROVED** [review]

**Pull request merged**